nomad/state/state_store.go

```diff
@@ -3924,6 +3923,36 @@ func (s *StateStore) SchedulerSetConfig(idx uint64, config *structs.SchedulerCon
 	return nil
 }
 
+func (s *StateStore) ClusterMetadata() (*structs.ClusterMetadata, error) {
+	txn := s.db.Txn(false)
+	defer txn.Abort()
+
+	// Get the cluster metadata
+	m, err := txn.First("cluster_meta", "id")
+	if err != nil {
+		return nil, fmt.Errorf("cluster metadata lookup failed: %v", err)
+	}
+
+	metadata, ok := m.(*structs.ClusterMetadata)
+	if !ok {
+		return nil, nil
```

Comment:
Is this effectively a nil check?  I'd suggest returning a proper error if the cluster metadata isn't of an expected type.
I see the pattern in other functions as well though.